### PR TITLE
Updated blocks

### PR DESCRIPTION
**Pull request by mosemister**
> Im new to documentation on public repos. 

This PR is designed to update the ``creating a plugin`` to API 8 in particular the ``blocks`` section.

### Concepts

Added Tags 

<details>
<summary>Concepts page screenshot</summary>

Concepts

</details>

### Accessing

Removed ``API 7``'s Block Property info as this is exclusively done with ``Keys`` in terms of accessing block data. The data manipulation in the ``data`` section will not be removed. 

Changed ``API 7``'s Block Trait to Block State Property.

<details>
<summary>Accessing page screenshot</summary>

Block/Accessing

</details>

### Modifying

<details>
<summary>Concepts page screenshot</summary>

Modifiying

</details>

### Block Entities

Changed the file name from tile entity to block entity

<details>
<summary>Concepts page screenshot</summary>

Block entities

</details>

### Virtual Block

There isnt much that needed to be changed here so I havent uploaded a image as its just import changes.

I have added a tip of ``BlockChangeFlags.NOTIFY_CLIENT`` too

### CODE REVIEW

File: source/plugin/blocks/accessing.rst [hunk 1]
```diff
@@ -2,217 +2,150 @@
...
+    org.spongepowered.api.world.server.ServerWorld
+    org.spongepowered.api.state.BooleanStateProperty
+    org.spongepowered.api.data.DataManipulator.Immutable
 
 Basic Information
 ~~~~~~~~~~~~~~~~~
 
 Blocks are most commonly identified and accessed by their :javadoc:`Location`. This location points to a certain
-coordinate within an :javadoc:`Extent`. In most cases a :javadoc:`World` will be used as the ``Extent``.
+coordinate within an :javadoc:`World`. In most cases a :javadoc:`ServerWorld` will be used as the ``World``.
```
Comment (Inscrutable) [inline source/plugin/blocks/accessing.rst]: Change `an` into `a`.

File: source/plugin/blocks/accessing.rst [hunk 2]
```diff
@@ -2,217 +2,150 @@
...
+    public boolean isBanner(ServerLocation blockLoc) {
+        Tag<BlockType> bannerTag = BlockTypeTags.BANNERS.get();
+        BlockType blockType = blockLoc.blockType();
+        return bannerTag.contains(blockType);
     }
 
 .. tip::
     
-    The function ``==`` could be used in place of ``equals()`` as there is only one :javadoc:`BlockType` instance for
+    The function ``==`` could be used in place of ``equals()`` when it comes to comparing two :javadoc:`BlockType`'s as there are a fixed number of instance for
```
Comment (Inscrutable) [inline source/plugin/blocks/accessing.rst]: instance -> instances. Also please try to keep line lengths shorter (> 120), it gets hard to proofread.

File: source/plugin/blocks/accessing.rst [hunk 3]
```diff
@@ -2,217 +2,150 @@
...
     every block, however it is generally recommended to use ``equals()``.
 
 Block Data Manipulators
 ~~~~~~~~~~~~~~~~~~~~~~~
 
 The data of a block is held as a :javadoc:`DataManipulator`, similar to other parts of the API. This is the container
-that holds information on components of our block such as the orientation of a block, specific types (stone vs.
-granite), and so on. Checking the values of these manipulators is easy, you just need to check the block's direction
-:javadoc:`DirectionalData`.
+that holds information on components of our block such as the orientation of a block, the blocks light level, and so on. Checking the values of these manipulators is easy, you just need to check the block's direction
```
Comment (Inscrutable) [inline source/plugin/blocks/accessing.rst]: Line length <120 please. 
This bit could also improve by being a bit shorter anyway; "the orientation of a block, the blocks light level, and so on." could be "the orientation, light level, and so on." (Implicitly they belong to our block).

File: source/plugin/blocks/accessing.rst [hunk 4]
```diff
@@ -2,217 +2,150 @@
...
     }
 
-First, we need to know which ``DataManipulator`` sub-interface we need. Those that are applicable to blocks are found
-in the :javadoc:`org.spongepowered.api.data.manipulator.mutable` and
-:javadoc:`org.spongepowered.api.data.manipulator.mutable.block` packages. Then, we can just pass that class to the
-``get(DataManipulator)`` method of ``Location`` which will return an ``Optional``. We then have to check if our
-``DataManipulator`` actually exists for our block by checking ``ifPresent()``. If it exists, then we can use it.
+First, we need to know which ``DataManipulator`` ``Key`` we need. We can then pass the ``get(Key)`` method of 
+``ServerLocation`` which will return an ``Optional``. We then have to check if our ``DataManipulator`` actually
+exists for out block by checking ``ifPresent()``. If it exists, then we can use it.
```
Comment (Inscrutable) [inline source/plugin/blocks/accessing.rst]: out -> our

**Review by Inscrutable — COMMENTED** [review]
Great Start! I only have a few minor quibbles, like keeping line length <120. 
References to other new pages can be inserted as they spawn. 
The page preview you pasted up is pretty cool too.

File: source/plugin/blocks/accessing.rst (hunk 4 above)
Comment (mosemister) [reply source/plugin/blocks/accessing.rst]: Typo. Now fixed

File: source/plugin/blocks/accessing.rst (hunk 3 above)
Comment (mosemister) [reply source/plugin/blocks/accessing.rst]: Removed ``block`` from the wording, also ensured line length

File: source/plugin/blocks/accessing.rst (hunk 2 above)
Comment (mosemister) [reply source/plugin/blocks/accessing.rst]: thanks, changed

File: source/plugin/blocks/accessing.rst (hunk 1 above)
Comment (mosemister) [reply source/plugin/blocks/accessing.rst]: thanks, changed

**Comment by mosemister** [comment]
Removed the draft status. Let the floor come at me ;-)

File: source/plugin/blocks/blockentities.rst [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+    org.spongepowered.api.block.entity.BlockEntity
+    org.spongepowered.api.block.entity.BlockEntityType
+    org.spongepowered.api.block.entity.carrier.CarrierBlockEntity
+    org.spongepowered.api.data.DataHolder
+    org.spongepowered.api.data.manipulator.DataManipulator
+    org.spongepowered.api.world.Location
+
+Block entities are blocks that are capable of additional operations like autonomically crafting (furnaces and brewing
+stands) or provide effects (like a beacon or a note block). They also contain additional data like an inventory or
+text (like chests, signs or command blocks).
```
Comment (Inscrutable) [inline source/plugin/blocks/blockentities.rst]: comma needed after signs

File: source/plugin/blocks/blockentities.rst [hunk 6]
```diff
@@ -0,0 +1,119 @@
...
+
+Identifying Block Entities and their Type
+=========================================
+
+Again, it all starts with a :javadoc:`Location`. The :javadoc:`Location#blockEntity()` function will return the tile
```
Comment (Inscrutable) [inline source/plugin/blocks/blockentities.rst]: I don't think the `Again` helps much here.

File: source/plugin/blocks/blockentities.rst [hunk 7]
```diff
@@ -0,0 +1,119 @@
...
+entity corresponding to the block or ``Optional.empty()`` if the block is not a tile entity.
```
Comment (Inscrutable) [inline source/plugin/blocks/blockentities.rst]: comma needed before `or`

File: source/plugin/blocks/blockentities.rst [hunk 8]
```diff
@@ -0,0 +1,119 @@
...
+ .. code-block:: java
+
+    import org.spongepowered.api.world.server.ServerLocation;
+
+    public boolean isTileEntity(ServerLocation blockLoc) {
+        return blockLoc.blockEntity().isPresent();
+    }
+
+The type of a block entity can then be obtained by the :javadoc:`BlockEntity#type()` function which returns a
+:javadoc:`BlockEntityType`. Which can then be compared similar to a :javadoc:`BlockType`. After performing this check
```
Comment (Inscrutable) [inline source/plugin/blocks/blockentities.rst]: `Which` is a bad way to start a sentence. In any case it needs to be a bit clearer, i.e.
`This can then be compared in a manner similar to a` [...]

File: source/plugin/blocks/blockentities.rst [hunk 9]
```diff
@@ -0,0 +1,119 @@
...
+            jukebox.eject();
+        }
+    }
+
+After performing this cast, the methods provided by the particular interface can be accessed (in this example the
+:javadoc:`Jukebox#eject()` method). For detailed information about ``BlockEntity`` subtypes and their respective
+methods refer to the :javadoc:`org.spongepowered.api.block.entity` package and its subpackages in the API.
+
+
+Accessing and Modifying a Block Entity's Data
```
Comment (Inscrutable) [inline source/plugin/blocks/blockentities.rst]: We could possibly shorten this to `Accessing and Modifying Block Entity Data`. 
In any case we need to be consistent with the use of the possessive (Baker's) apostrophe. Avoid it where possible.

File: source/plugin/blocks/blockentities.rst [hunk 10]
```diff
@@ -0,0 +1,119 @@
...
+=============================================
+
+Similar to block states, the data stored in a tile entity is accessed using a :javadoc:`DataManipulator`. Since the
+kind of a data is fully described by the ``DataManipulator`` used, all data manipulation can be done with the
```
Comment (Inscrutable) [inline source/plugin/blocks/blockentities.rst]: kind -> type
data is a plural, so drop the `a`

File: source/plugin/blocks/blockentities.rst [hunk 11]
```diff
@@ -0,0 +1,119 @@
...
+    }
+
+The main difference to working with a ``BlockState`` is that a block entity is a mutable :javadoc:`DataHolder` as
+opposed to the immutable ``BlockState``.
+
+Accessing Inventories
+=====================
+
+Quite a share of block entities come with their own inventory, most notably chests and furnaces. That inventory cannot
+be accessed directly from the ``BlockEntity`` interface. So a cast will be necessary. Since all block entities containing
```
Comment (Inscrutable) [inline source/plugin/blocks/blockentities.rst]: Tack the smaller sentence onto the prior one, i.e. `[...] interface, so a cast will be necessary.`

File: source/plugin/blocks/blockentities.rst [hunk 12]
```diff
@@ -0,0 +1,119 @@
...
+an inventory extend the :javadoc:`CarrierBlockEntity` interface it suffices to cast to that interface as shown below.
```
Comment (Inscrutable) [inline source/plugin/blocks/blockentities.rst]: comma needed between `interface` and `it`

File: source/plugin/blocks/modifying.rst [hunk 13]
```diff
@@ -42,77 +45,58 @@ Altering Block States
...
     }
 
-Note that the :javadoc:`DirtData` is a mutable copy of the data held in the ``BlockState``. It is changed and then
-converted back to an immutable and used to create a new ``BlockState`` which then replaces the original block.
-
 The ``without()`` method accepts a class reference and will create a new ``BlockState`` without the data
 represented by the given class. If the block state would not be valid without that data, a default value will be used.
-So if the ``DirtData`` from a dirt blocks state is removed, it will fall back to :javadoc:`DirtTypes#DIRT`, the default
-value. The following example will dry the block at a given ``Location``, if possible.
+So if the ``Keys.DIRECTION`` from a blocks state is removed, it will fall back to :javadoc:`Direction#NORTH`,
```
Comment (Inscrutable) [inline source/plugin/blocks/modifying.rst]: from a `block's` state

File: source/plugin/blocks/modifying.rst [hunk 14]
```diff
@@ -42,77 +45,58 @@ Altering Block States
...
         if (dryState.isPresent()) {
             blockLoc.setBlock(dryState.get());
         }
     }
 
-Since the :javadoc:`WetData` data manipulator represents boolean data, by removing it we set the wetness of the block
-(if it has any) to false. The ``dryState.isPresent()`` check will fail on block states that cannot be wet since
-``dryState`` will be ``Optional.empty()`` in that case.
+Since the :javadoc:`Keys#IS_WET` data manipulator represents boolean data, by removing it we set the wetness of the block
+(if it has any) to false. The ``withEastState.isPresent()`` check will fail on block states that cannot be wet since
```
Comment (Inscrutable) [inline source/plugin/blocks/modifying.rst]: East?

**Review by Inscrutable — COMMENTED** [review]
Loads of minor quibbles and an unexpected appearance of East. Lots of love for Baker's apostrophes. Good going.

File: source/plugin/blocks/modifying.rst (hunk 14 above)
Comment (mosemister) [reply source/plugin/blocks/modifying.rst]: dryState ;)

File: source/plugin/blocks/modifying.rst (hunk 13 above)
Comment (mosemister) [reply source/plugin/blocks/modifying.rst]: updated

File: source/plugin/blocks/blockentities.rst (hunk 12 above)
Comment (mosemister) [reply source/plugin/blocks/blockentities.rst]: updated

File: source/plugin/blocks/blockentities.rst (hunk 11 above)
Comment (mosemister) [reply source/plugin/blocks/blockentities.rst]: updated

File: source/plugin/blocks/blockentities.rst (hunk 10 above)
Comment (mosemister) [reply source/plugin/blocks/blockentities.rst]: updated

File: source/plugin/blocks/blockentities.rst (hunk 9 above)
Comment (mosemister) [reply source/plugin/blocks/blockentities.rst]: updated

File: source/plugin/blocks/blockentities.rst (hunk 8 above)
Comment (mosemister) [reply source/plugin/blocks/blockentities.rst]: updated

File: source/plugin/blocks/blockentities.rst (hunk 7 above)
Comment (mosemister) [reply source/plugin/blocks/blockentities.rst]: updated

File: source/plugin/blocks/blockentities.rst (hunk 6 above)
Comment (mosemister) [reply source/plugin/blocks/blockentities.rst]: updated

File: source/plugin/blocks/blockentities.rst (hunk 5 above)
Comment (mosemister) [reply source/plugin/blocks/blockentities.rst]: updated

File: source/plugin/blocks/blockentities.rst [hunk 15]
```diff
@@ -56,11 +56,11 @@ After performing this cast, the methods provided by the particular interface can
 methods refer to the :javadoc:`org.spongepowered.api.block.entity` package and its subpackages in the API.
 
 
-Accessing and Modifying a Block Entity's Data
+Accessing and Modifying Block Entity's Data
 =============================================
```
Comment (Inscrutable) [inline source/plugin/blocks/blockentities.rst]: Better fix the underline length too or the parser will bit3 you
